lib/jekyll-remote-theme/gemspec_manager.rb

```diff
@@ -0,0 +1,58 @@
+# frozen_string_literal: true
+
+module Jekyll
+  module RemoteTheme
+    # Since parsing / evaluating a gemspec from third-party themes can lead to arbitrary
+    # code execution, this class exists to read the given file and handle the contents
+    # within.
+    class GemspecManager
+      SKIP_LINE_REGEX = %r!\A(?:\s*(?:#|\$|require)|\n)!
+      DEPENDENCY_MATCHER = %r!add_[runtime_]*dependency!
+      DEPENDENCY_CAPTURE_REGEX = %r!#{DEPENDENCY_MATCHER}\(?\s*(?<dependency>[^)#]+)\s*!
+      EXTRACT_QUOTED_STRING_REGEX = %r!(["'])(.*?[^\\])\1!
+
+      # Returns the trimmed contents of the provided gemspec file.
+      attr_reader :spec_contents
+
+      # Returns an array of dependencies, each entry being a sub-array containing the
+      # gem-name and its required version information.
+      attr_reader :spec_dependencies
+
+      def initialize(gemspec)
+        @gemspec = gemspec
+        @spec_contents = []
+        @spec_dependencies = []
+
+        read
```

Comment:
Since `#read` is definitely necessary, I elected to have it inside `initialize` and `private :read`  in lieu of using guard clauses to ensure memoized content.